src/WinRT.Runtime/Projections/Nullable.cs

```diff
@@ -2203,6 +2006,139 @@ internal static class Nullable_Delegates
+                return __retval;
+            }
+            finally
+            {
+                Marshal.Release(nullablePtr);
+            }
+        }
+    }
+
+    internal static class NullableType
+    {
+        internal static Guid GetIIDForBuiltInType(global::System.Type type)
+        {
+            if (type == typeof(int)) return Nullable_int.IID;
+            if (type == typeof(byte)) return Nullable_byte.IID;
+            if (type == typeof(bool)) return Nullable_bool.IID;
+            if (type == typeof(sbyte)) return Nullable_sbyte.IID;
+            if (type == typeof(short)) return Nullable_short.IID;
+            if (type == typeof(ushort)) return Nullable_ushort.IID;
+            if (type == typeof(char)) return Nullable_char.IID;
+            if (type == typeof(uint)) return Nullable_uint.IID;
+            if (type == typeof(long)) return Nullable_long.IID;
+            if (type == typeof(ulong)) return Nullable_ulong.IID;
+            if (type == typeof(float)) return Nullable_float.IID;
+            if (type == typeof(double)) return Nullable_double.IID;
+            if (type == typeof(Guid)) return Nullable_guid.IID;
+            if (type == typeof(global::System.Type)) return Nullable_Type.IID;
+            if (type == typeof(global::System.TimeSpan)) return Nullable_TimeSpan.IID;
+            if (type == typeof(global::System.DateTimeOffset)) return Nullable_DateTimeOffset.IID;
+            if (type == typeof(global::Windows.Foundation.Point)) return IReferenceIIDs.IReferenceOfPoint_IID;
+            if (type == typeof(global::Windows.Foundation.Size)) return IReferenceIIDs.IReferenceOfSize_IID;
+            if (type == typeof(global::Windows.Foundation.Rect)) return IReferenceIIDs.IReferenceOfRect_IID;
+            if (type == typeof(global::System.Numerics.Matrix3x2)) return IReferenceIIDs.IReferenceMatrix3x2_IID;
+            if (type == typeof(global::System.Numerics.Matrix4x4)) return IReferenceIIDs.IReferenceMatrix4x4_IID;
+            if (type == typeof(global::System.Numerics.Plane)) return IReferenceIIDs.IReferencePlane_IID;
+            if (type == typeof(global::System.Numerics.Quaternion)) return IReferenceIIDs.IReferenceQuaternion_IID;
+            if (type == typeof(global::System.Numerics.Vector2)) return IReferenceIIDs.IReferenceVector2_IID;
+            if (type == typeof(global::System.Numerics.Vector3)) return IReferenceIIDs.IReferenceVector3_IID;
+            if (type == typeof(global::System.Numerics.Vector4)) return IReferenceIIDs.IReferenceVector4_IID;
+
+            return default;
+        }
+
+        public static Func<IInspectable, object> GetValueFactory(global::System.Type type)
+        {
+            return ComWrappersSupport.CreateReferenceCachingFactory(GetValueFactoryInternal(type));
+        }
+
+        private static Func<IInspectable, object> GetValueFactoryInternal(global::System.Type type)

```

Comment:
Hey @MichalStrehovsky, any ideas on how we could change this method to avoid to always unconditionally root all of those generic instantiations even if the types for those checks have been trimmed out? Right now this will otherwise always bring in like 35 KB worth of stuff that you might not even be using at all 🥲

Reply:
If a `System.Type` is all you have, you can't avoid it.

We could optimize this out at the codegen phase because we know this once we have the whole program view. It's a JitInterface change and I've been avoiding those because they're painful. I might look at it because this is not the first time...

Optimizing it out during codegen is not panacea because whatever is under the `if` branch could still cause damage to whole program view that cannot be undone during codegen anymore (like seeing the delegate construction if the app also uses `Delegate.Method` with that delegate) but maybe it will be all that's needed here.